__init__.py

```diff
@@ -106,40 +113,93 @@ def respond(self, query):
         if len(query) == 0:
             return 0.0
 
-        r = ddg.query(query)
+        # note: '1+1' throws an exception
+        try:
+            r = ddg.query(query)
+        except:
+            LOG.warning("DDG exception!")
+            return None
 
-        LOG.debug('Query: ' + str(query))
-        LOG.debug('Type: ' + r.type)
+        LOG.debug("Query: %s" % (str(query),))
```

Comment:
While % formatting is still supported, consider using an updated string formatting method.

Reply:
The [logger module is optimized for % formatting](https://docs.python.org/3/howto/logging.html#optimization). It's not going to have a wild impact in this one instance but I wonder if it's worth using it by default for consistency?

We could make it a little cleaner though eg:
```
LOG.debug("Query: %s", str(query))
```

Reply:
Pretty sure if I touched the line it does not use the '+' syntax as I believe this can cause undesired exceptions.